smithy-model/src/main/java/software/amazon/smithy/model/pattern/SmithyPattern.java

```diff
@@ -111,6 +113,43 @@ public boolean equals(Object other) {
         return other instanceof SmithyPattern && pattern.equals(((SmithyPattern) other).pattern);
     }
 
+    /**
+     * Gets a list of explicitly conflicting label segments between this
+     * pattern and another.
+     *
+     * @param otherPattern SmithyPattern to check against.
+     * @return A list of Segment Pairs where each pair represents a conflict
+     *     and where the left side of the Pair is a segment from this pattern.
+     */
+    public List<Pair<Segment, Segment>> getConflictingLabelSegments(SmithyPattern otherPattern) {
```

Comment:
Can we just return a map and state that the map maintains order?

Reply:
Yes, but this was an exiting method from UriPattern so it's gonna get ugly. Pushed the change in a separate commit, enjoy